### ISSUE TITLE
Fix SAWarning in logs

### ISSUE THREAD

**Issue by thomasmorgan** [Dallinger/Dallinger#269]
When sandboxing an experiment the logs are full of the following error:

```
Oct 19 11:50:50 dlgr-c81320c6 app/clock.1:  /app/.heroku/python/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:1034: SAWarning: Reassigning polymorphic association for identity 'memory_gene' from <Mapper at 0x7fe87ccfc310; MemoryGene> to <Mapper at 0x7fe87ccb4e50; MemoryGene>: Check for duplicate use of 'memory_gene' as value for polymorphic_identity. 
```

These occur for all the custom classes defined by the experiment, they also occur on all dynos (web, worker, clock).

We suspect that the offending code is this:

```
# Load the experiment.
try:
    exp = imp.load_source('experiment', "dallinger_experiment.py")
    classes = inspect.getmembers(exp, inspect.isclass)
    exps = [c for c in classes
            if (c[1].__bases__[0].__name__ in "Experiment")]
    this_experiment = exps[0][0]
    mod = __import__('dallinger_experiment', fromlist=[this_experiment])
    experiment = getattr(mod, this_experiment)

except ImportError:
    print("Error: Could not import experiment.")
```

which is in custom.py (web dyno) and heroku/clock.py (clock dyno). Its not in heroku/worker.py but the worker functions are defined in custom, so that is a potential route by which the worker dyno could also produce these errors.

Finally, this error does not seem to cause any issues, but it does clutter up the logs.
## Expected Behavior

Error should not be in logs
## Current Behavior

Error appears in logs
## Steps to Reproduce (for bugs)

run any experiment in the sandbox, play it.

**Comment by suchow** [comment]
I'm also seeing this in debug mode on `development`.

**Comment by suchow** [comment]
Okay, made some progress in figuring this out. It's happening because we're importing the specific experiment module several times: once in the command line's `verify` command to check if there's an Experiment class present, once in the config function that registers extra settings, and once to load the experiment for real. What we need is some way to tell SQLAlchemy to ignore things on the second and later imports (or, more generally, on all but one of them).

**Comment by jessesnyder** [comment]
The warning is raised by the presence of a class definition for a SQLAlchemy-derived class in the same module as the experiment's main `Experiment` subclass (in the case of the Bartlett1932 demo, it’s the `WarOfTheGhostsSource` class).  SQLAlchemy assumes that ORM-derived classes are only loaded once, and it relies on Python’s module import caching to enforce this.  Because of the special way the dallinger framework pulls in the `experiment` module from the experiment working directory, `imp.load_module()` is used in place of a traditional import. `load_module` does not consult the module cache; it forces a reload every time it is called, and we call it 4 times.

As far as I can tell, the warning raised by SQLAlchemy is directed at developers who have created more than one class with the same value for `polymorphic_identity`, either because they copied and pasted to create a new class and forgot to change it, or subclassed and didn’t override the value in the subclass. In the case addressed by this ticket, since it is always the same class being re-assigned under the same mapping key, there no impact beyond triggering the warning message. (SQLAlchemy doesn’t check to see if the value is changing or not, it just notices that the key has already be registered and is now being re-registered.)

I considered trying to pull the experiment module from the `sys.modules` caching instead of calling `imp.load_module()`, but this felt like adding more magic, so I didn't pursue it very far.

Moving the ORM class out of the module that holds the special `Experiment` subclass resolves the problem, with a minor(?) complication: The ORM “helper” class cannot be imported when the `experiment` module is loaded, as far as I can tell. An absolute path import isn’t an option, and a relative import doesn’t work because the modules are not part of a package. Therefore, the import has to happen at runtime, finding the module by virtue of it being located in the current working directory. The fact that import cannot use a package namespace to limit the context of the matches increases the risk of name collisions. To protect against this in the Bartlett1932 example, I named the new helper module `bartlett1932_source.py`, so the import statement is `from bartlett1932_source import WarOfTheGhostsSource`. Despite this shortcoming, to me it seems like keeping as much code as possible *outside* of the `experiment.py` module makes sense, given the way it is loaded by the framework.

(see https://github.com/Dallinger/Dallinger/tree/issues/269-sqla-reimport)

**Comment by suchow** [comment]
Closed by #466.
